src/game/boe.main.cpp

```diff
@@ -440,33 +440,28 @@ static void handle_scenario_args() {
 		}
 		// Try to put the party in an outdoor section from which you can enter the town --
 		// so when you leave, you'll hopefully be in the right place.
-		bool found_entrance = false;
-		for(int x = 0; x < univ.scenario.outdoors.width(); ++x){
-			for(int y = 0; y < univ.scenario.outdoors.height(); ++y){
-				for(spec_loc_t& entrance : univ.scenario.outdoors[x][y]->city_locs){
-					if(entrance.spec == *scen_arg_town){
-						// Very janky but I don't know how else to make it properly load the right sections and set i_w_c
-						while(univ.party.outdoor_corner.x > x){
-							shift_universe_left();
-						}
-						while(univ.party.outdoor_corner.x < x){
-							shift_universe_right();
-						}
-						while(univ.party.outdoor_corner.y > y){
-							shift_universe_up();
-						}
-						while(univ.party.outdoor_corner.y < y){
-							shift_universe_down();
-						}
-						outd_move_party(local_to_global(entrance), true);
-
-						found_entrance = true;
-						break;
-					}
-				}
-				if(found_entrance) break;
+		auto town_entrances = univ.scenario.find_town_entrances(*scen_arg_town);
+		if(!town_entrances.empty()){
+			// TODO a dialog could be shown to choose between multiple entrances,
+			// but maybe that would be janky, because this happens when you're trying to launch
+			// from INSIDE a town.
```

Comment:
No need. If there are multiple entrances, the town should already be set up to properly handle them – if they are in different outdoor sections, there would be a "change outdoor location" special node near each viable exit, for example. And if they're using the "set town exits in town settings" feature, that should also correctly reposition them when they leave town.

So, this code only really matters if there is just one town entrance… though it certainly doesn't hurt to take the first if there are multiple, just in case.